ttnn/cpp/ttnn/operations/transformer/attention_softmax/attention_softmax.cpp

```diff
@@ -26,26 +26,34 @@ ttnn::Tensor ExecuteAttentionSoftmax<in_place>::invoke(
     } else {
         if (not attention_mask.has_value()) {
             auto output_tensor = ttnn::multiply(input_tensor, head_size);
-            return ttnn::operations::normalization::softmax(output_tensor, memory_config.value_or(input_tensor.memory_config()));
+            return ttnn::prim::softmax(
+                ttnn::operations::normalization::SoftmaxDeviceOperation::operation_attributes_t{
+                    .memory_config = memory_config.value_or(output_tensor.memory_config()),
+                },
+                ttnn::operations::normalization::SoftmaxDeviceOperation::tensor_args_t{
+                    .input_tensor = output_tensor,
+                }
```

Comment:
instead of passing operation_args and tensor_args, this method accepts normal arguments and packs them to those args structures for internal use.

Reply:
in a way, they should not leak to an operation

Reply:
oh, at the same time

https://github.com/tenstorrent/tt-metal/pull/11953/files